### PR TITLE
[connector/routing] warn if a routing table entry already exists

### PR DESCRIPTION
**Pull request by kristofgyuracz**
**Description:** <Describe what has changed.>
Based on discussion in #30663, a warning is logged if there are two or more routing items with the same routing statement.

**Link to tracking Issue:** #30663

### CODE REVIEW

File: connector/routingconnector/router.go [hunk 1]
```diff
@@ -119,6 +119,8 @@ func (r *router[C]) registerRouteConsumers() error {
 		route, ok := r.routes[key(item)]
 		if !ok {
 			route.statement = statement
+		} else {
+			r.logger.Warn("statement already exists in the routing table, the existing routing item will be overwritten")
```
Comment (jpkrohling) [inline connector/routingconnector/router.go:123]: Be nice to the user and tell them which items is being used :-)

Comment (kristofgyuracz) [reply connector/routingconnector/router.go]: yep, it does.

File: connector/routingconnector/router.go [hunk 2]
```diff
@@ -120,7 +121,12 @@ func (r *router[C]) registerRouteConsumers() error {
 		if !ok {
 			route.statement = statement
 		} else {
-			r.logger.Warn("statement already exists in the routing table, the existing routing item will be overwritten")
+			pipelineNames := []string{}
+			for _, pipeline := range item.Pipelines {
+				pipelineNames = append(pipelineNames, pipeline.Name())
+			}
+			exporters := strings.Join(pipelineNames, ", ")
+			r.logger.Warn(fmt.Sprintf(`statement "%s" already exists in the routing table, the route with exporters %s will be ignored`, item.Statement, exporters))
```
Comment (jpkrohling) [inline connector/routingconnector/router.go]:
```suggestion
			r.logger.Warn(fmt.Sprintf(`statement %q already exists in the routing table, the route with exporters %q will be ignored`, item.Statement, exporters))
```

**Review by jpkrohling — APPROVED** [review]
LGTM. I have a small suggestion, but feel free to ignore it.

File: connector/routingconnector/router.go (hunk 2 above)
Comment (kristofgyuracz) [reply connector/routingconnector/router.go]: The suggestion was based on an earlier commit, recommitted it. Thanks for the quick feedback. :)

**Review by jpkrohling — APPROVED** [review]

**Comment by kristofgyuracz** [comment]
Is this a change that should be added to the changelog?
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#:~:text=Refactorings%20with%20no%20meaningful%20change%20in%20functionality

**Comment by jpkrohling** [comment]
Yes

**Review by mwear — APPROVED** [review]

**Pull request merged**